### PR TITLE
Add command based logic to motorZPP

### LINKED ISSUE #19: [Bug] * on Motor ZPP not showing
[AB#26605](https://dev.azure.com/praxis-tech/c6eef564-6c10-48c8-ad5d-8c1c9bc170aa/_workitems/edit/26605)

## Problem
A case has been submitted where a `*` is expected in the **Motor ZPP**.

When a value tagged with a `*` directly affects the calculation, the **Motor ZPP** needs to be flagged with a `*`.

Cases:

### Case 1

<figure>
  Error case image
  <figcaption>ISNCSCI example 1 update. From Rudiger. November 2022. Wrong Motor ZPP. Right should be T7*</figcaption>
</figure>

### LINKED ISSUE #21: [Bug] Wrong Motor ZPP
[AB#23838](https://dev.azure.com/praxis-tech/c6eef564-6c10-48c8-ad5d-8c1c9bc170aa/_workitems/edit/23838)

## Problem

A case was reported where the Motor ZPP is incorrect.

On the right, the `0` Pin Prick values at `T4`, `T5`, and `T6` break the continuity therefore making the right ZPP `T3`.

On the left, the `0` at `T11` pin prick also breaks the continuity making the left ZPP `T10`.

<figure>
  Error case image
  <figcaption>Shepherd MA. Wrong Motor ZPP. August 2022, from Mary. Should be <b>Right T3</b> and <b>Left T10</b>.</figcaption>
</figure>

### Related cases

<figure>
  Error case image
  <figcaption>Example 1: Left Motor ZPP should be <b>T2</b></figcaption>
</figure>

---

<figure>
  Error case image
  <figcaption>Example 2: Right and left M ZPP should be <b>T2</b></figcaption>
</figure>

---

<figure>
  Error case image
  <figcaption>Example 3: Left Motor ZPP should be <b>T2</b></figcaption>
</figure>

### LINKED ISSUE #23: [Bug] Motor ZPP only scoring as `NA` when VAC is set to `NT`
[AB#15071](https://dev.azure.com/praxis-tech/c6eef564-6c10-48c8-ad5d-8c1c9bc170aa/_workitems/edit/15071)

## Problem

Existing case 112 is reporting Motor ZPP value of `[NA]` for a for with all normal motor and sensory values (5s and 2s) but with `VAC = NT`.

The expected result is `[NA,S3]` as the Motor ZPP would return `[NA]` when `VAC = Yes` and `[S3]` when `VAC = No`.

### PR DESCRIPTION
**Pull request by EddieMachete**
Depends on #25 
Closes #19, #21, #23

## Problem

1. There are issues tracking the continuity of sensory function when calculating the Motor ZPP scores producing errors in the calculation:
   - #19
   - #21 
   - #23 
2. It is hard to follow the code and understand where the issue is making it hard for a developer to fix any logic issues.

## Solution

Instead of writing the calculation as a loop, I am rewriting the Motor ZPP calculation with a _command like_ structure where we have a function for every step in the calculation. Each step returns the next step. This will help us:

1. Make the code more legible
2. Be able to step through an individual calculation to see what the algorithm is doing

- We maintain a state throughout the calculation.
- Each command function receive the state as a parameter and return a `Step` object which contains the description and text defining the actions that took place during the execution of the command (useful for reporting and training), a new and updated `State` object, and the next command function to be executed.

## Testing instructions

I suggest that for testing the approver relies on the automated test cases from file `specs/2019.json`:
- Related to issue #19, we added the new test case [121]
- For issue #21, we added test cases [122-125]
- Issue #23 is related to test case [112]

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -2,7 +2,8 @@
   "compilerOptions": {
     "resolveJsonModule": true,
     /* Basic Options */
-    "target": "ES3",                          /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017', 'ES2018', 'ES2019' or 'ESNEXT'. */
+    "target": "ES5",                          /* Specify ECMAScript target version: 'ES5' (default), 'ES2015', 'ES2016', 'ES2017', 'ES2018', 'ES2019' or 'ESNEXT'. */
```
Comment (EddieMachete) [inline tsconfig.json:5]: `Target ES3` is deprecated so I've upgraded to `ES5`. This also allows us to use newer array methods like `includes`.

File: specs/2019.json [hunk 2]
```diff
@@ -17471,7 +17471,7 @@
       "rightSensoryZpp":"T1",
       "leftSensoryZpp":"S1",
       "rightMotorZpp":"C8*",
-      "leftMotorZpp":"L4*,L5*"
+      "leftMotorZpp":"L4,L5*"
```
Comment (EddieMachete) [inline specs/2019.json:17474]: This case broke after the logic was updated. `L4` should not have been flagged with `*` as no `*` value affects it. I've reviewed this case with Kris Walden.

File: specs/2019.json [hunk 3]
```diff
@@ -18290,8 +18290,8 @@
       "neurologicalLevelOfInjury": "INT",
       "rightSensoryZpp": "NA",
       "leftSensoryZpp": "NA",
-      "rightMotorZpp": "NA",
-      "leftMotorZpp": "NA",
+      "rightMotorZpp": "NA,S3",
```
Comment (EddieMachete) [inline specs/2019.json:18293]: Related to #23

File: specs/2019.json [hunk 4]
```diff
@@ -18622,7 +18622,7 @@
       "asiaImpairmentScale":"B",
       "rightSensoryZpp":"NA",
       "leftSensoryZpp":"NA",
-      "rightMotorZpp":"T12",
+      "rightMotorZpp":"T12*",
```
Comment (EddieMachete) [inline specs/2019.json:18625]: After updating the logic, this case started breaking. Values should have a `*`. I reviewed this case with Kris Walden.

**Review by praxis-tech — APPROVED** [review]
LGTM!
